realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -183,8 +192,25 @@ class RealmTransformer extends Transform {
                     }
                 }
             }
-        }
 
+            it.jarInputs.each {
+                def jarFile = new JarFile(it.file)
+                def entries = jarFile.entries()
+                entries.each {
```

Comment:
This could be more compact and readable using `filter` IMO:

```java
entries
   .filter {
      !it.directory && it.path.endsWith(SdkConstants.DOT_CLASS)
   }
   .each {
      def className = path.substring(0, path.length() - SdkConstants.DOT_CLASS.length()).replace(File.separatorChar, '.' as char)
      classNames.add(className)
   }
```